test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/LiabilityLiableDateControllerSpec.scala

```diff
@@ -0,0 +1,179 @@
+          mockRegistrationUpdate(aRegistration())
+
+          val correctForm = Seq("answer" -> "yes", formAction)
+          val result      = controller.submit()(postJsonRequestEncoded(correctForm: _*))
+
+          status(result) mustBe SEE_OTHER
+          modifiedRegistration.liabilityDetails.isLiable mustBe Some(true)
+
+          formAction._1 match {
+            case "SaveAndContinue" =>
+              redirectLocation(result) mustBe Some(
+                routes.CheckLiabilityDetailsAnswersController.displayPage().url
+              )
+            case "SaveAndComeBackLater" =>
+              redirectLocation(result) mustBe Some(routes.RegistrationController.displayPage().url)
+          }
+        }
+
+        "user submits 'No' answer" in {
+          authorizedUser()
+          mockRegistrationFind(aRegistration())
+          mockRegistrationUpdate(aRegistration())
+
+          val correctForm = Seq("answer" -> "no", formAction)
+          val result      = controller.submit()(postJsonRequestEncoded(correctForm: _*))
+
+          status(result) mustBe SEE_OTHER
+
+          modifiedRegistration.liabilityDetails.isLiable mustBe Some(false)
+
+          formAction._1 match {
+            case "SaveAndContinue" =>
+              redirectLocation(result) mustBe Some(routes.NotLiableController.displayPage().url)
+            case "SaveAndComeBackLater" =>
+              redirectLocation(result) mustBe Some(routes.RegistrationController.displayPage().url)
+          }
+        }
+      }
+
+      "return 400 (BAD_REQUEST) for " + formAction._1 when {
+        "user does not pick an answer" in {
+          authorizedUser()
+          mockRegistrationFind(aRegistration())
+          val result =
+            controller.submit()(postRequestEncoded(JsObject.empty, formAction))
+
+          status(result) mustBe BAD_REQUEST
+        }
+
+        "user enters invalid data" in {
+          authorizedUser()
+          mockRegistrationFind(aRegistration())
+          val incorrectForm = Seq("answer" -> "maybe", formAction)
+          val result        = controller.submit()(postJsonRequestEncoded(incorrectForm: _*))
+
+          status(result) mustBe BAD_REQUEST
+        }
+      }
+
+      "return an error for " + formAction._1 when {
+
+        "user is not authorised" in {
+          unAuthorizedUser()
+          val result = controller.displayPage()(getRequest())
+
+          intercept[RuntimeException](status(result))
+        }
+
+        "user submits form and the registration update fails" in {
+          authorizedUser()
+          mockRegistrationFind(aRegistration())
+          mockRegistrationUpdateFailure()
+
+          val correctForm = Seq("answer" -> "yes", formAction)
+          val result      = controller.submit()(postJsonRequestEncoded(correctForm: _*))
+
+          intercept[DownstreamServiceError](status(result))
+        }
+
+        "user submits form and a registration update runtime exception occurs" in {
```

Comment:
Does this test basically cover when HTTP codes such as `500`, etc. are returned?

Reply:
yes